service/mongodb/store.go

```diff
@@ -212,6 +218,43 @@ func (s MongoStore) GetAll(ctx context.Context, filter interface{}) ([]service.P
 	return permissions, nil
 }
 
+// GetUserPermissionsByPage returns a slice of the permissions requested by the filter,
+// in the page they belong to by page number and page size.
+func (s MongoStore) GetUserPermissionsByPage(ctx context.Context, pn int64, ps int64, sortBy bson.D, filter interface{}) (*PagingRes, error) {
+	collection := s.DB.Collection(PermissionCollectionName)
+
+	itemCount, err := collection.CountDocuments(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := options.Find().SetSort(sortBy).SetLimit(ps).SetSkip(pn * ps)
+
+	cur, err := collection.Find(ctx, filter, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	permissions := []service.Permission{}
+	for cur.Next(ctx) {
```

Comment:
a cursor is not thread safe and should only be accessed by a single go-routine at any given time. you should add - ` defer cur.Close(ctx)` before iterate with next. (you can see the implementation at  GetAll)

Reply:
good comment :+1: